### PR TITLE
[master] an aptpkg major cleanup for 3008.x

### LINKED ISSUE #65703: [BUG] pkgrepo.managed inserts malformed 'signedby' string in generated source file
**Description**
Debian based source.list files have you optionally specify a "signed-by" key in brackets to the absolute path of a gpg key. currently experiencing all of my salt `pkgrepo.managed` that previously worked are now inserting "signedby" text in the brackets. Looking at the `pkgrepo.py` modules it looks like this isnt happening here but in another salt module it uses and most likely a line parsing mistake is happening.

**SLS**
```yaml
# setting official postgres repository
ensure_postgres_repository_present:
  pkgrepo.managed:
    - humanname: Postgresql direct database repo
    - name: |
        deb [signed-by=/etc/apt/keyrings/postgresql_signing.key] 
        https://apt.postgresql.org/pub/repos/apt {{ grains.oscodename }}-pgdg main 
    - file: /etc/apt/sources.list.d/postgresql.list
    - keyurl: https://www.postgresql.org/media/keys/ACCC4CF8.asc
    - aptkey: False
    - disabled: False
    - refresh: True
    - architectures: amd64
```
**state return**
```
          ID: ensure_postgres_repository_present
    Function: pkgrepo.managed
        Name: deb [signed-by=/etc/apt/keyrings/postgresql_signing.key] 
https://apt.postgresql.org/pub/repos/apt jammy-pgdg main 

      Result: False
     Comment: Failed to configure repo 'deb [signed-by=/etc/apt/keyrings/postgresql_signing.key] 
              https://apt.postgresql.org/pub/repos/apt jammy-pgdg main 
              ': E: Malformed entry 1 in list file /etc/apt/sources.list.d/postgresql.list ([option] not assignment)
              E: The list of sources could not be read.
     Started: 16:26:22.904497
    Duration: 58.133 ms
     Changes:   
```
**resulting file**
```
deb [signed-by=/etc/apt/keyrings/postgresql_signing.key signedby] https://apt.postgresql.org/pub/repos/apt jammy-pgdg main
```
The "signedby" in the brackets shouldnt be there. This happens for my nginx, docker, and postgres salt states that have worked for a long time prior to this. 

- [x] on-prem machine
- [ ] VM (Virtualbox, KVM, etc. please specify)
- [x] VM running on a cloud service, please be explicit and add details
- [ ] container (Kubernetes, Docker, containerd, etc. please specify)
- [ ] or a combination, please be explicit
- [ ] jails if it is FreeBSD
- [ ] classic packaging
- [x] onedir packaging
- [ ] used bootstrap to install

**Steps to Reproduce the behavior**
create a `pkgrepo.managed` state not using `aptkey` or `PPA` and set `refresh: True` to trigger apt refresh. 

**Expected behavior**
I expected the repo files to successfully be created resulting in a follow up installation from another salt state.

**Screenshots**

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.)

```yaml
Salt Version:
          Salt: 3006.5
 
Python Version:
        Python: 3.10.13 (main, Nov 15 2023, 04:34:27) [GCC 11.2.0]
 
Dependency Versions:
          cffi: 1.14.6
      cherrypy: unknown
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.1.2
       libgit2: Not Installed
  looseversion: 1.0.2
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     packaging: 22.0
     pycparser: 2.21
      pycrypto: Not Installed
  pycryptodome: 3.9.8
        pygit2: Not Installed
  python-gnupg: 0.4.8
        PyYAML: 6.0.1
         PyZMQ: 23.2.0
        relenv: 0.14.2
         smmap: Not Installed
       timelib: 0.2.4
       Tornado: 4.5.3
           ZMQ: 4.3.4
 
System Versions:
          dist: ubuntu 22.04.3 jammy
        locale: utf-8
       machine: x86_64
       release: 6.2.0-1012-aws
        system: Linux
       version: Ubuntu 22.04.3 jammy
```
</details>

**Additional context**

### PR DESCRIPTION
**Pull request by whytewolf**
### What does this PR do?
This is a major overhaul of `aptpkg` for 3008. remove system libraries we do not have access for in relenv or virtualenvs which is the major push forward. cleaning up some of the code as well as introduce a pkg.which function which follows the FreeBSD version to allow people to find which package installed a file. 

### What issues does this PR fix or reference?
Fixes: #65703 #66056 #66201 

### Previous Behavior
it works, mostly.

### New Behavior
it works better. 

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [X] Docs
- [X] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [X] Tests written/updated

### CODE REVIEW

File: changelog/66056.changed.md [hunk 1]
```diff
@@ -0,0 +1 @@
+re-work the aptpkg module to remove system libraries that onedir and virtualenvs do not have access too. too streamline testing, and code use to needed libraries only.
```
Comment (twangboy) [inline changelog/66056.changed.md:1]: `too. too`

**Review by twangboy — CHANGES_REQUESTED** [review]

**Comment by twangboy** [comment]
Looks like you need a rebase

**Review by twangboy — DISMISSED** [review]

**Comment by whytewolf** [comment]
> Looks like you need a rebase

conflict fixed.

**Review by twangboy — DISMISSED** [review]

**Review by twangboy — DISMISSED** [review]

**Comment by twangboy** [comment]
Looks like you got some pre-commit failures

**Comment by whytewolf** [comment]
> Looks like you got some pre-commit failures
yeah i was working on those as you typed that. took a while for pre-commit to finish.

File: tests/pytests/functional/utils/pkg/test_deb.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            "result": {
+                "signedby": {
+                    "full": "signed-by=/etc/apt/keyrings/example.key",
+                    "value": "/etc/apt/keyrings/example.key",
+                },
+                "arch": {"full": "arch=amd64", "value": ["amd64"]},
+            },
+        },
+        {
+            "oneline": "deb [arch=amd64 signed-by=/etc/apt/keyrings/example.key]  https://example.com/pub/repos/apt xenial main",
```
Comment (dmurphy18) [inline tests/pytests/functional/utils/pkg/test_deb.py:18]: Wondering if we should also test with arm64 too here, to ensure we don't miss something there.

**Review by dmurphy18 — COMMENTED** [review]
Wondering about tests with arm64 as an architecture test

File: tests/pytests/functional/utils/pkg/test_deb.py (hunk 2 above)
Comment (whytewolf) [reply tests/pytests/functional/utils/pkg/test_deb.py:18]: I can add that. but with what the function actually does it won't change anything. it doesn't do value checking on those values. or actually connect to any repos. it just pulls the variables out and apart.

Comment (dmurphy18) [reply tests/pytests/functional/utils/pkg/test_deb.py:18]: Saw that , just wanted an arm64 test in there too, to catch anyone in the future mucking up and making something Intel specific, so more a catch of future changes than yours which look fine.

Comment (whytewolf) [reply tests/pytests/functional/utils/pkg/test_deb.py:18]: alright, added. to both this and SourceEntry.repo_line tests.

**Review by twangboy — APPROVED** [review]

**Review by dmurphy18 — APPROVED** [review]

**Review by dwoz — APPROVED** [review]

**Pull request merged**